lib/parser/parse-tree.h

```diff
@@ -889,9 +889,10 @@ struct TypeParamDecl {
 //        integer-type-spec , type-param-attr-spec :: type-param-decl-list
 // R734 type-param-attr-spec -> KIND | LEN
 struct TypeParamDefStmt {
-  ENUM_CLASS(KindOrLen, Kind, Len)  // R734
   TUPLE_CLASS_BOILERPLATE(TypeParamDefStmt);
-  std::tuple<IntegerTypeSpec, KindOrLen, std::list<TypeParamDecl>> t;
+  std::tuple<IntegerTypeSpec, common::TypeParamKindOrLen,
```

Comment:
Can Kind and Length ever be used interchangeably?  If not, shouldn't this be a variant of either a Kind or a Length instead of using the enum to discriminate between the two?

Reply:
All the enum represents here is this reduction: `type-param-attr-spec -> KIND | LEN`
There is no data associated with it, so I don't think there is any value in using a variant.